accepted/future-releases/0546-patterns/feature-specification.md

```diff
@@ -2832,7 +2844,7 @@ To match a pattern `p` against a value `v`:
     3.  Check the length. If `p` is empty or has any non-rest elements:
```

Comment:
This "If" clause irks me slightly. Not entirely sure why, maybe it's the lack of obvious logical connection between the two clauses. So I start thinking "why?" when I read it.

We're describing three cases out of four combinations of "has rest elements" and "has non-rest pattern".
The case we don't need to include is "has rest pattern and has no non-rest pattern".

The description is actually spot-on for that, maybe its just the wrong way around.
Consider: "If `p` has any non-rest elements or `p` has no rest elements" (where the latter, guarded by the former, is the same as being empty).

(Overthinking much? Nah!)

Also, should "rest elements" be "rest pattern" or "rest elements pattern"?

Reply:
I think the current wording might be marginally better?